WordPress/Classes/ViewRelated/Blog/Blog + Me/UIBarButtonItem+MeBarButton.swift

```diff
@@ -1,39 +1,22 @@
 import Gridicons
 import UIKit
 
-
 /// Add a UIBarButtonItem to the navigation bar that  presents the Me scene.
-extension BlogDetailsViewController {
-    @objc
-    func presentHandler() {
-        meScenePresenter.present(on: self, animated: true, completion: nil)
-    }
-
+extension UIViewController {
     @objc
-    func addMeButtonToNavigationBar() {
-        navigationItem.rightBarButtonItem = UIBarButtonItem(email: blog.account?.email,
-                                                            target: self,
-                                                            action: #selector(presentHandler))
+    func addMeButtonToNavigationBar(email: String?, meScenePresenter: ScenePresenter) {
+        navigationItem.rightBarButtonItem = UIBarButtonItem(
+            email: email,
+            action: {
+                meScenePresenter.present(on: self, animated: true, completion: nil)
```

Comment:
We need a weak self here, or we end up with a retain cycle on BlogDetailViewController.

Reply:
'Tis done.

If you're happy with the PR feel free to approve, I'll wait for @Gio2018 's review anyway since I'm modifying some of his code.